File: Java/Search/BJAlgo_1543.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class BJAlgo_1543 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String string = br.readLine();
        String word = br.readLine();

        int count = 0;
        int findIdx = 0;

        while (string.length() - word.length() >= findIdx) {
            if (string.substring(findIdx, findIdx + word.length()).equals(word)) {
                count += 1;
                findIdx += word.length();
            } else {
                findIdx += 1;
            }
        }

        System.out.println(count);
    }
}

File: Python/Search/BJAlgo_1543.py
from sys import stdin

string = stdin.readline().strip()
search = stdin.readline().strip()

count = 0
find_idx = 0
while find_idx <= len(string) - len(search):
    if string[find_idx:find_idx + len(search)] == search:
        count += 1
        find_idx += len(search) + 1
    else:
        find_idx += 1

print(count)
